cardano-api/src/Cardano/Api/Orphans.hs

```diff
@@ -686,3 +688,37 @@ instance Crypto.Crypto crypto => ToJSON (VMap VB VB (Shelley.KeyHash    'Shelley
 instance Crypto.Crypto crypto => ToJSON (VMap VB VP (Shelley.Credential 'Shelley.Staking   crypto) (Shelley.CompactForm Shelley.Coin)) where
   toJSON = toJSON . fmap fromCompact . VMap.toMap
   toEncoding = toEncoding . fmap fromCompact . VMap.toMap
+
+-----
+
+instance ToJSON (Consensus.StakeSnapshots crypto) where
+  toJSON = object . stakeSnapshotsToPair
+  toEncoding = pairs . mconcat . stakeSnapshotsToPair
+
+stakeSnapshotsToPair :: Aeson.KeyValue a => Consensus.StakeSnapshots crypto -> [a]
```

Comment:
I think this logic shouldn't be in the JSON instance but at the point of rendering. The JSON instance should only be concerned with rendering JSON. In the cli we should determine how we render depending on if we get a `Map` of length 1 or more.

Reply:
There's not a good way to do the fact there is a `Map` at all breaks backwards compatibility.

This PR is meant to preserve backwards compatibility.

A follow up PR simplifies this code but breaks backwards compatibility.  https://github.com/input-output-hk/cardano-node/pull/4279

It would be good to get an optimised version of this command merged first before merging the change that breaks compatibility.

Reply:
I've just re-discovered `data Stakes`. This is what we were rendering. What we should do to maintain backwards compatibility is create a function with type signature `Consensus.StakeSnapshots crypto -> Stakes`.  Imposing our cli logic on a JSON instance seems wrong, which should just be concerned with rendering JSON.

Reply:
The data doesn't fit into `data Stakes` either because the map can be empty and I would not have anything to fill the pool fields in that case.

I've rendered the JSON manually instead.